### ISSUE TITLE
gno test log output is buffered, and joined for subtests

### ISSUE THREAD

**Issue by thehowl** [gnolang/gno#1081]
At the time being, test output is buffered, even when the test has subtests (note: only applies to t.Log functions, and others from `testing.T` -- println and fmt.PrintXX work normally). Furthermore, when there are subtests, the output of these is joined together.

```
=== RUN   TestPerft
=== RUN   TestPerft/n0
=== RUN   TestPerft/n1
=== RUN   TestPerft/n2
=== RUN   TestPerft/n3
=== RUN   TestPerft/n4
=== RUN   TestPerft/n5
=== RUN   TestPerft/n6
--- PASS: TestPerft (247.57s)
output: counts: [20 400 8902]

counts: [48 2039 97862]

counts: [14 191 2812 43238]

counts: [6 264 9467]

counts: [6 264 9467]

counts: [44 1486 62379]

counts: [46 2079]
ok      ./.     249.56s
```

```go
// example minimal test to reproduce
package main

import (
        "testing"
)

func long() {
        // (sleep-like)
        c := func(x int) {}
        for i := 0; i < (1 << 20); i++ {
                c(i)
        }
}

func TestHelloWorld(t *testing.T) {
        t.Log("Test log")
        long()
        t.Log("Test log")
        long()
        t.Log("Test log")
}
```

This makes for poor UX as:

- For long running tests, they are seemingly doing nothing until they are finished
- The behaviour is different from what one might expect with Go
- The final buffered output does not offer a way to distinguish from which subtest the output actually came from.

_(Part of a series of issues encountered while developing [gnochess](https://github.com/gnolang/gnochess))_

**Comment by salihdhaifullah** [comment]
I tried to reproduce this and found out that `t.Log` only works when verbose flag is on, is this a bug ?

**Comment by thehowl** [comment]
@salihdhaifullah It's not, it's intended behaviour :) (for when tests succeed)

This is actually fixed by #1594, I forgot this issue existed.
